snow/engine/common/bootstrapper.go

```diff
@@ -116,76 +79,11 @@ func (b *bootstrapper) GetAcceptedFrontierFailed(ctx context.Context, nodeID ids
 		return nil
 	}
 
-	if !b.pendingReceiveAcceptedFrontier.Contains(nodeID) {
-		b.Ctx.Log.Debug("received unexpected GetAcceptedFrontierFailed message",
-			zap.Stringer("nodeID", nodeID),
-		)
-		return nil
-	}
-
-	// If we can't get a response from [nodeID], act as though they said their
-	// accepted frontier is empty and we add the validator to the failed list
-	b.failedAcceptedFrontier.Add(nodeID)
-	return b.markAcceptedFrontierReceived(ctx, nodeID)
-}
-
-func (b *bootstrapper) markAcceptedFrontierReceived(ctx context.Context, nodeID ids.NodeID) error {
-	// Mark that we received a response from [nodeID]
-	b.pendingReceiveAcceptedFrontier.Remove(nodeID)
-
-	b.sendGetAcceptedFrontiers(ctx)
-
-	// still waiting on requests
-	if b.pendingReceiveAcceptedFrontier.Len() != 0 {
-		return nil
-	}
-
-	// We've received the accepted frontier from every bootstrap validator
-	// Ask each bootstrap validator to filter the list of containers that we were
-	// told are on the accepted frontier such that the list only contains containers
-	// they think are accepted.
-	totalSampledWeight, err := b.sampledBeacons.TotalWeight(b.Ctx.SubnetID)
-	if err != nil {
-		return fmt.Errorf("failed to get total weight of sampled beacons for subnet %s: %w", b.Ctx.SubnetID, err)
-	}
-	beaconsTotalWeight, err := b.Beacons.TotalWeight(b.Ctx.SubnetID)
-	if err != nil {
-		return fmt.Errorf("failed to get total weight of beacons for subnet %s: %w", b.Ctx.SubnetID, err)
-	}
-	newAlpha := float64(totalSampledWeight*b.Alpha) / float64(beaconsTotalWeight)
-
-	failedBeaconWeight, err := b.Beacons.SubsetWeight(b.Ctx.SubnetID, b.failedAcceptedFrontier)
-	if err != nil {
-		return fmt.Errorf("failed to get total weight of failed beacons: %w", err)
-	}
-
-	// fail the bootstrap if the weight is not enough to bootstrap
-	if float64(totalSampledWeight)-newAlpha < float64(failedBeaconWeight) {
-		if b.Config.RetryBootstrap {
-			b.Ctx.Log.Debug("restarting bootstrap",
-				zap.String("reason", "not enough frontiers received"),
-				zap.Int("numBeacons", b.Beacons.Count(b.Ctx.SubnetID)),
-				zap.Int("numFailedBootstrappers", b.failedAcceptedFrontier.Len()),
-				zap.Int("numBootstrapAttemps", b.bootstrapAttempts),
-			)
-			return b.Restart(ctx, false)
-		}
-
-		b.Ctx.Log.Debug("didn't receive enough frontiers",
-			zap.Int("numFailedValidators", b.failedAcceptedFrontier.Len()),
-			zap.Int("numBootstrapAttempts", b.bootstrapAttempts),
-		)
-	}
```

Comment:
This whole mechanism was very questionable. The security isn't impacted by the frontier... The check at the end that the accepted frontier is non-empty should be enough to ensure that we don't mark bootstrapping as finished before we should. Technically there could be some weird situation where we only end up getting a majority of responses for something like the genesis block... But this is really a hacky attempt to prevent that at best.

I think it makes sense to just remove it.